Init colorama in ESPHome main

kuba2k2 opened the pull request:
# What does this implement/fix?

<!-- Quick description and explanation of changes -->

Call `colorama.init()` in ESPHome CLI tools, to fix terminal color support on Windows (especially in editors such as VSCode, having their own terminal support).

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

kuba2k2 committed:
Init colorama in ESPHome main

jesserockz commented:
Which part exactly is this fixing?
Do you have a before and after screenshot?

kuba2k2 commented:
This fixes colored text output, where otherwise ANSI color codes are shown instead.

Take a look (before/after):
![obraz](https://github.com/esphome/esphome/assets/30433568/44f4d7c5-328b-4637-935e-5bf68f03980a)

(the 2nd one errors out, because it doesn't have the Popen() PR checked out).

Nevertheless, it's always good to init Colorama whenever working with colored text output.

jesserockz commented:
Weird, it was colouring for me when I was testing the popen PR just fine on windows.

Since it is going to be init'd here, can you remove the init from the log file?

kuba2k2 committed:
Remove colorama.init() from log.py

kuba2k2 commented:
Sure, I didn't really notice it was in `log.py` already. It seems that the whole issue was that `init()` was placed *after* the `logging` calls - placing it before them fixes coloring.

Anyway, I removed it from `log.py`, and moved to `run_esphome()` since it seems more appropriate.

jesserockz commented:
If moving it before the `logging` calls fixes the issue on windows, I would rather keep in inside the log.py file in that case...
(sorry to go around in circles)

kuba2k2 committed:
Move colorama.init() back to log.py

kuba2k2 commented:
Done.

jesserockz commented:
Thanks!
